internal/server/models/destination.go

```diff
@@ -26,7 +26,7 @@ func (d *Destination) ToAPI() *api.Destination {
 	connected := false
 	// TODO: this should be configurable
 	// https://github.com/infrahq/infra/issues/2505
-	if time.Since(d.LastSeenAt) < 5*time.Minute {
+	if time.Since(d.LastSeenAt) < 5*time.Minute && d.ConnectionURL != "" {
```

Comment:
Does this mean pending destinations never timeout and disconnect?

Reply:
That's right. Destinations will be in the Pending state until they report a connection URL. This might change as I can see it leading to some confusion. Instead, the status should be 1) Disconnected if not seen within last X minutes, 2) Pending if seen but there's no connection string, and 3) Connected if seen and has a connection string. Will update the UI as well.